### DISCUSSION TITLE
Connecting Wemos D1 to USB

### DISCUSSION THREAD

**Discussion by odroegehorn** [syssi/esphome-soyosource-gtn-virtual-meter#95] [General]
Hi all,
great project.
How can I connect a WeMos D1 mini not only to RS485 (with adapter), but also to the USB Port of the SoyoSource?
Just plain pin-by-pin? 
Or only 5v/GND to the WeMos and RX/TX via resistors?
We have most probably level differences between 3.3V / 5V ?

I have seen the YAML file for this setup, so it seems to be possible.

Thanks for explanation.

**Comment by syssi** [comment]
I'm using a level shifter to shift the logic levels. I assume a voltage divider using some resistors should work too.

**Comment by pfeupfeu** [comment]
There was a discussion if the ESP2866 are 5V-tolerant at their GPIOs. It seems, they are. 
So I connected the USB Port of the SoyoSource via 3k9 resistors. It works flawlessy.
